packages/xchain-thorchain/src/client.ts

```diff
@@ -84,8 +82,26 @@ class Client implements ThorchainClient, XChainClient {
     return this.network
   }
 
+  setClientUrl = (clientUrl: string): void => {
+    this.clientUrl = clientUrl
+    this.thorClient = this.getNewThorClient()
+  }
+
   getClientUrl = (): string => {
-    return this.network === 'testnet' ? 'http://18.198.92.45:1317' : 'http://138.68.125.107:1317'
+    return this.clientUrl
+  }
+
+  getDefaultClientUrl = (): string => {
```

Comment:
In addition to previous comment, return of `{testnetClientUrl: string, mainnetClientUrl: string}` is needed then. 

And you may need to have another function `getClientUrlByNetwork = (network: Network): string`.